scripts/install.sh

```diff
@@ -0,0 +1,772 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# Do not call this script within your cloned git repo.
+# This script will be called by the user directly from an external source (eg., internet link)
+
+# If there's an existing installation in the same location,
+# - If the script's Mathesar's version = installed version, it performs all operations without affecting the installation.
+#   - Useful if the user is facing environment related issues and would like to get them fixed.
+#   - Running & re-running this script on a working Mathesar installation would have no effect on Mathesar itself.
+# - If the script's Mathesar version > installed version, it updates the installed version with the one on this script.
+# - If the script's Mathesar version < installed version, it throws an error and stops.
+
+# If there's an existing installation in a different location, it shows a message and prompts the user to continue.
+
+# The resulting directory structure would be as follows:
+#
+# ── mathesar/ (parent installation directory - must exist)
+#    ├── bin/
+#    │   ├── mathesar                   (main executable bash script)
+#    │   ├── mathesar_path_source       (path setting script for sh-like shells)
+#    │   └── mathesar_path_source.fish  (path setting script for fish - if fish is installed)
+#    ├── uv/
+#    ├── packages/
+#    ├── mathesar-venv/
+#    ├── .env
+#    ├── .media
+#    ├── config/
+#    ├── db/
+#    ├── mathesar/
+#    └── ... (other source files)
+
+
+#=======CONFIGURATIONS=========================================================
+
+MATHESAR_VERSION="0.2.3"
```

Comment:
I don't like maintaining this in multiple places. Can it be injected during the build?

Reply:
Agreed, I don't like it too. I wanted to do this incrementally, but I'll try to get the changes for it done in this PR.

Reply:
I've made the necessary changes:
- mathesar script: Takes the version from mathesar/__init__.py
- install script: Gets injected during build